### ISSUE TITLE
Remove 'document.body.contains' check for container

### ISSUE THREAD

**Issue by bisubus** [janpaepke/ScrollMagic#470]
Can we remove [document.body.contains](https://github.com/janpaepke/ScrollMagic/blob/v2.0.5/dev/src/ScrollMagic/Controller.js#L81) check?

It causes the controller to silently fall back to `window` for detached containers (I've had some tough time tearing`window` changes down in tests because of that, but the same situation can take place in production code). The same thing will also happen with shadow DOM container, which is an evident bug.

**Comment by janpaepke** [comment]
Sorry for the extremely long wait on this. Good catch -- the `document.body.contains` check silently falling back to `window` for detached containers and shadow DOM was a real bug. It made testing harder and could cause unexpected behavior in production with web components.

v3's architecture handles this differently. Each ScrollMagic instance accepts a `scrollParent` option for specifying the scroll container, and it does not fall back silently. Shadow DOM support would depend on IntersectionObserver's `root` option, which works with shadow DOM in modern browsers. If you encounter issues with shadow DOM in v3, a new issue would be welcome.

---
**ScrollMagic v3 is now available** — a complete rewrite with an entirely new architecture and API. v2 will not receive further updates (2.0.9 is the final release).

👉 `npm install scrollmagic@next` — [Docs](https://github.com/janpaepke/ScrollMagic/blob/main/README.md) · [Release notes](https://github.com/janpaepke/ScrollMagic/releases/tag/v3.0.0-beta.0) · [npm](https://www.npmjs.com/package/scrollmagic/v/3.0.0-beta.0)
